### PR TITLE
Lavaland Golems

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
More exciting respawn adventures (though without explicit hostility this time).

:cl: Kor
rscadd: There are a couple new lavaland ruins, including one with a new ghost role.
/:cl:

### CODE REVIEW

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 1]
```diff
@@ -245,3 +245,77 @@
...
+/obj/effect/mob_spawn/human/golem
+	name = "completed golem shell"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "construct"
+	mob_species = /datum/species/golem
+	roundstart = FALSE
+	death = FALSE
+	anchored = 0
+	density = 0
+	flavour_text = {"<B>You are a Free Golem. Your family worships <span class='danger'>The Liberator</span>. In his infinite and divine wisdom, he set your clan free to travel the stars with a single declaration; "Yeah go do whatever." Though you are bound to the one who created you, it is customary in your society to repeat those same words to newborn golems, so that no golem may ever be forced to serve again.</B>"}
```
Comment (KazeEspada) [inline code/modules/ruins/lavaland_ruin_code.dm]: whatever = GREEF

**Comment by Supermichael777** [comment]
looks like the machine you put it as being built in is somehow invalid.
check to see if it runtimes with any other design because its logic looks a little weird.

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 2]
```diff
@@ -245,3 +245,77 @@
...
+
+/datum/design/golem_shell
+	name = "Golem Shell Construction"
+	desc = "Allows for the construction of a Golem Shell."
+	id = "golem"
+	req_tech = list("materials" = 12)
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 20000)
+	build_path = /obj/item/golem_shell
+	category = list("Misc")
```
Comment (Supermichael777) [inline code/modules/ruins/lavaland_ruin_code.dm]: i dont think this is a valid catagory

Comment (KorPhaeron) [inline code/modules/ruins/lavaland_ruin_code.dm]: Nah it wasn't the category, your first reply helped me figure out what was wrong though, thanks

File: code/modules/ruins/lavaland_ruin_code.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/ruins/lavaland_ruin_code.dm]: also you need to escape that set of "

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 3]
```diff
@@ -245,3 +245,81 @@
...
+/obj/effect/mob_spawn/human/golem
+	name = "completed golem shell"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "construct"
+	mob_species = /datum/species/golem
+	roundstart = FALSE
+	death = FALSE
+	anchored = 0
+	density = 0
+	flavour_text = {"<B>You are a Free Golem. Your family worships <span class='danger'>The Liberator</span>. In his infinite and divine wisdom, he set your clan free to travel the stars with a single declaration; 'Yeah go do whatever.' Though you are bound to the one who created you, it is customary in your society to repeat those same words to newborn golems, so that no golem may ever be forced to serve again.</B>"}
```
Comment (Cheridan) [inline code/modules/ruins/lavaland_ruin_code.dm]: :laughing:

Comment (KazeEspada) [inline code/modules/ruins/lavaland_ruin_code.dm]: TIME2GRIFF

**Comment by RandomMarine** [comment]
Can golems withstand ash storms?

**Comment by KorPhaeron** [comment]
IDs dont work in machine/cant be forged cause not traitors

completed golem shells should send a ghost message or be visually distinct

doors are kind of weird with smooth walls

you can get stacks of 0 metal if you use a stack of 10 on the golem shell

**Comment by KorPhaeron** [comment]
@RandomMarine 

Turns out no

**Pull request merged**